### DISCUSSION TITLE
Inconsistency in Naming of ClientWebSocketResponse.reader

### DISCUSSION THREAD

**Discussion by goznauk** [aio-libs/aiohttp#7923] [Q&A]
Hello,

I'm writing to express a concern about the ClientWebSocketResponse class, specifically regarding its 'reader' and 'writer' members. I noticed that while 'reader' is typed as "FlowControlDataQueue[WSMessage]", 'writer' is defined as "WebSocketWriter".

The 'WebSocketReader' class can be instantiated with a 'FlowControlDataQueue' for the 'read_queue' and an integer 'max_msg_size'. It is then injected into the 'ClientWebSocketResponse' through the 'response.connection.protocol.set_parser' of a 'ClientResponse' object.

Within the 'ClientWebSocketResponse', there seems to be a predominant use of 'FlowControlDataQueue', which is also accessible through 'WebSocketReader.queue'.

This leads me to wonder why the type of 'ClientWebSocketResponse.reader' is 'FlowControlDataQueue'. Although I am relatively new to this field, it appears more logical to me that it would be of the 'WebSocketReader' type.

Considering this, would it be more appropriate to access this via 'self.reader.queue' instead of just 'self.reader'? I am curious if there is a specific reason for the current implementation that I might not be aware of. Additionally, if changing it as suggested seems beneficial, I am willing to contribute to making this modification.

Thank you.

**Comment by Dreamsorcerer** [comment]
> This leads me to wonder why the type of 'ClientWebSocketResponse.reader' is 'FlowControlDataQueue'. Although I am relatively new to this field, it appears more logical to me that it would be of the 'WebSocketReader' type.

I'm not entirely sure what the confusion is. It is a FlowControlDataQueue, created at:
https://github.com/aio-libs/aiohttp/blob/213d1b22d42ce8efc54d7858b490e920fcdb4f0a/aiohttp/client.py#L887
And passed in at:
https://github.com/aio-libs/aiohttp/blob/213d1b22d42ce8efc54d7858b490e920fcdb4f0a/aiohttp/client.py#L902-L903

In general, if you have any more questions like this, I'd suggest you try changing the annotation and then run `mypy`. The errors that are produced should make it clear why the change can't work. If you still have questions after that, then please also link to the parts of the code that you are talking about, as it will make it a lot easier to understand what you're looking at.
